docs/ADR-58-third-party-assets-integration-v2-with-merkle-tree.md

```diff
@@ -0,0 +1,180 @@
+- **Committee Members (curators)**: the committee members are going to be validated by querying the committee smart contract directly. Committee members can approve/reject third parties and their items. They can also review third parties items with one simple transaction by submitting a merkle tree root and process items consumption signatures from third parties managers.
+- **Third Party Managers**: the third party managers are a set of addresses that can add items to the third party record, previously added by someone on the committee, and update the third party and items metadata. Also, they sign messages to consume items. The message and the signature must be submitted by a member of the committee after checking if it is genuine.
+
+#### Third Party Record
+
+```solidity
+struct ThirdParty {
+        bool isApproved;
+        bytes32 root; // new
+        uint256 maxItems;
+        uint256 consumedSlots; // new
+        uint256 registered;
+        string metadata;
+        string resolver;
+        string[] itemIds;
+        mapping(bytes32 => uint256) receipts; // new
+        mapping(address => bool) managers;
+        mapping(string => Item) items;
+        mapping(string => bool) rules; // new
+    }
+```
+
+- _`isApproved`_: whether a third party is approved or not.
+- _`root`_: current merkle tree root. This root is going to be used by the Decentraland catalyst to check if an item deployment is valid or not.
+- _`maxItems`_: represents the maximum number of items that a third party can have. We call them _item slots_. Item slots can be bought by everyone at any time, on multiple occasions, by using MANA. So, it is not necessary to be a third party manager to buy item slots. Item slots are going to be bought by unit. The price is set in USD dollar defined in the _`itemSlotPrice`_ variable.
+- _`consumedSlots`_: the number of items consumed by a third party.
+- _`registered`_: simple boolean that helps to check whether a third party has been added or not.
+- _`metadata`_: string with the following shape: `type:version:name:description`. i.e: `tp:1:third party 1:the third party 1 description`. We may include collections metadata by changing the metadata version.
+- _`resolver`_: string with the third party API resolver. This API will be used for services to get which Decentraland asset should be mapped to which NFT token. _We call Decentraland asset to every asset that is submitted to the Decentraland catalyst_. i.e: `https://api.thirdparty1.com/v1/get-owned-nfts/:owner`
+- _`itemIds`_: in order to allow looping through the items added without the need of indexing historic events, we need to keep track of their ids. **IMPORTANT**: Items reviewed by using a merkle tree root, are not going to be submitted to the blockchain.
+- _`receipts`_: mapping of item consumptions messages hash. We will use this in order to prevent double-spending them.
```

Comment:
What's the `bytes32` that the map receives?

Reply:
The `messageHash`.